pilot/pkg/bootstrap/server.go

```diff
@@ -211,6 +215,16 @@ func NewServer(args *PilotArgs, stopCh <-chan struct{}) (*Server, error) {
 		return nil, fmt.Errorf("sidecar injector: %v", err)
 	}
 
+	// Run the SDS signing server.
+	// RunCA() must be called after createCA() and initDNSListener()
+	// because it depends on the following conditions:
+	// 1) CA certificate has been created.
+	// 2) grpc server has been generated.
+	s.addStartFunc(func(stop <-chan struct{}) error {
+		s.RunCA(s.secureGRPCServerDNS, s.ca, caOpts, stop)
```

Comment:
if kubeclient is nil the s.ca will be nil here, which might behave bad

Reply:
Good point! A check of nil CA is added into RunCA().

Reply:
If the client is nil, then `s.ca = s.createCA(s.kubeClient.CoreV1(), caOpts)` will panic. Which seems impossible as protected by `EnableCA()`

Reply:
If the kubeClient is nil, s.ca = s.createCA(s.kubeClient.CoreV1(), caOpts) will NOT panic because when kubeClient is nil EnableCA() returns false and createCA will not be called.